pakyow-core/lib/core/middleware/req_path_normalizer.rb

```diff
@@ -0,0 +1,27 @@
+require 'core/call_context'
+
+module Pakyow
+  module Middleware
+    # Rack compatible middleware that normalize the path if contains '//',
+    # it replace '//' with '/' and issue a 301 redirect to the new path.
+    #
+    # @api public
+    class ReqPathNormalizer
+      def initialize(app)
+        @app = app
+      end
+
+      def call(env)
+        if env['PATH_INFO'].include? '//'
+          CallContext.new(env).redirect(normalized_path(env['PATH_INFO']), 301)
```

Comment:
@karimmtarek I pulled it down and can reproduce. It might be easier to test with a single case, something like `it 'issues a 301 redirect with the normalized path`.

```ruby
it 'issues a 301 redirect with the normalized path' do
  expect_any_instance_of(Pakyow::CallContext).to receive(:redirect).with(...)
  instance.call(env)
end
```

This is only one possible way to approach; you might have a better idea 😀

Maybe this helps? We could pair sometime.